python/ray/data/read_api.py

```diff
@@ -397,6 +398,96 @@ def read_parquet(
     )
 
 
+@PublicAPI(stability="alpha")
+def read_images(
+    paths: Union[str, List[str]],
+    *,
+    filesystem: Optional["pyarrow.fs.FileSystem"] = None,
+    parallelism: int = -1,
+    partition_filter: Optional[
+        PathPartitionFilter
+    ] = ImageDatasource.file_extension_filter(),
+    partitioning: Partitioning = None,
+    size: Optional[Tuple[int, int]] = None,
+    mode: Optional[str] = None,
+):
+    """Read images from the specified paths.
+
+    .. warning::
+        If your dataset contains images of varying sizes and you don't specify
+        ``size``, this function will error. To prevent errors, specify ``size``
+        or :ref:`disable tensor extension casting <disable_tensor_extension_casting>`.
```

Comment:
We can remove this limitation once ragged tensor support is re-merged: https://github.com/ray-project/ray/pull/29071